truffleruby/src/main/java/org/truffleruby/core/binding/BindingNodes.java

```diff
@@ -50,6 +51,36 @@
 @CoreClass("Binding")
 public abstract class BindingNodes {
 
+    @ImportStatic(BindingNodes.class)
+    public abstract static class CreateBindingNode extends RubyBaseNode {
+
+        public abstract DynamicObject execute(MaterializedFrame frame);
+
+        @Specialization(guards = "descriptor == frame.getFrameDescriptor()", limit = "getContext().getOptions().FRAME_VARIABLE_ACCESS_LIMIT")
+        public DynamicObject createBinding(MaterializedFrame frame,
+                @Cached("frame.getFrameDescriptor()") FrameDescriptor descriptor,
+                @Cached("newFrameDescriptor(getContext())") FrameDescriptor bindingDescriptor) {
```

Comment:
This will effectively cache a descriptor for multiple bindings created from the same frame.
I think it is unfortunately not correct currently.
```ruby
def m
  b1=binding
  b2=binding
  [b1,b2]
end

b1, b2 = m

p b1.local_variables
p b2.local_variables

b1.local_variable_set :e, 42
p b1.local_variables
p b2.local_variables

p b1.local_variable_get :e
p b2.local_variable_get :e
```

```bash
$ ruby binding.rb
[:b1, :b2]
[:b1, :b2]
[:e, :b1, :b2]
[:b1, :b2]
42
binding.rb:17:in `local_variable_get': local variable `e' not defined for #<Binding:0x0055bb3dd8fd58> (NameError)
	from binding.rb:17:in `<main>'
$ jt ruby binding.rb
[:b1, :b2]
[:b1, :b2]
[:e, :b1, :b2]
[:e, :b1, :b2]
42
nil
```